### PR TITLE
10 shopinvader product name

### PR DESCRIPTION
**Pull request by bguillot**
Adds the possibility to use a specific name for the product in shopinvader

### CODE REVIEW

File: shopinvader/models/shopinvader_product.py [hunk 1]
```diff
@@ -33,12 +33,27 @@ class ShopinvaderProduct(models.Model):
     url_key = fields.Char(
         compute_sudo=True,
     )
+    use_product_shopinvader_name = fields.Boolean(
+        related='backend_id.use_product_shopinvader_name',
+        store=True, readonly=True)
+    shopinvader_name = fields.Char(
+        string='Name',
+        help="Name for shopinvader, if not set the product name will be used.")
+    model_name = fields.Char(compute='_compute_name', readonly=True)
```
Comment (lmignon) [inline shopinvader/models/shopinvader_product.py]: @bguillot This name `model_name` is really confusing. In my mind, model always refers to a model into odoo...
why not `display_name` or `label` or ???

**Review by florian-dacosta — APPROVED** [review]

File: shopinvader/models/shopinvader_backend.py [hunk 2]
```diff
@@ -63,6 +63,11 @@ class ShopinvaderBackend(models.Model):
     filter_ids = fields.Many2many(
         comodel_name='product.filter',
         string='Filter')
+    use_product_shopinvader_name = fields.Boolean(
```
Comment (lmignon) [inline shopinvader/models/shopinvader_backend.py]:
```suggestion
    use_shopinvader_product_name = fields.Boolean(
```

File: shopinvader/models/shopinvader_backend.py [hunk 3]
```diff
@@ -63,6 +63,11 @@ class ShopinvaderBackend(models.Model):
...
+        string="Use product Shopinvader name",
```
Comment (lmignon) [inline shopinvader/models/shopinvader_backend.py]: Use Shopinvader product display name

File: shopinvader/models/shopinvader_backend.py [hunk 4]
```diff
@@ -63,6 +63,11 @@ class ShopinvaderBackend(models.Model):
...
+        help="If checked, use the field shopinvader name for product "
```
Comment (lmignon) [inline shopinvader/models/shopinvader_backend.py]: If checked, use the specific shopinvader display name for products

File: shopinvader/views/shopinvader_variant_view.xml [hunk 5]
```diff
@@ -9,7 +9,7 @@
         <field name="model">shopinvader.variant</field>
         <field name="arch" type="xml">
             <search string="Products">
-                <field name="name" string="Product"/>
+                <field name="name" string="Product" filter_domain="['|','|','|',('default_code','ilike',self),('name','ilike',self),('barcode','ilike',self),('shopinvader_name', 'ilike', self)]"/>
```
Comment (lmignon) [inline shopinvader/views/shopinvader_variant_view.xml]: /s/filter_domain/domain

IMO it's better to specify the domain into the field definition to ease the override (as a @api.model method)

**Pull request merged**